cpp/src/io/csv/csv_reader.cu

```diff
@@ -686,45 +689,50 @@ gdf_error read_csv(csv_read_arg *args)
 	free(h_dtypes); 
 	free(h_valid); 
 	free(h_data); 
-	
-	launch_dataConvertColumns(raw_csv, d_data, d_valid, d_dtypes, d_str_cols, d_valid_count);
-	cudaDeviceSynchronize();
 
-	stringColCount=0;
-	for (int col = 0; col < raw_csv->num_active_cols; col++) {
+	if (raw_csv->num_records != 0) {
+		error = launch_dataConvertColumns(raw_csv, d_data, d_valid, d_dtypes, d_str_cols, d_valid_count);
+		if (error != GDF_SUCCESS) {
+			return error;
+		}
+		cudaDeviceSynchronize();
 
-		gdf_column *gdf = cols[col];
+		stringColCount=0;
+		for (int col = 0; col < raw_csv->num_active_cols; col++) {
 
-		if (gdf->dtype != gdf_dtype::GDF_STRING)
-			continue;
+			gdf_column *gdf = cols[col];
 
-		NVStrings* const stringCol = NVStrings::create_from_index(h_str_cols[stringColCount],size_t(raw_csv->num_records));
-		if ((raw_csv->quotechar != '\0') && (raw_csv->doublequote==true)) {
-			// In PANDAS, default of enabling doublequote for two consecutive
-			// quotechar in quote fields results in reduction to single
-			std::string quotechar = std::string(&raw_csv->quotechar);
-			std::string doublequotechar = quotechar + raw_csv->quotechar;
-			gdf->data = stringCol->replace(doublequotechar.c_str(), quotechar.c_str());
-			NVStrings::destroy(stringCol);
-		}
-		else {
-			gdf->data = stringCol;
-		}
+			if (gdf->dtype != gdf_dtype::GDF_STRING)
+				continue;
 
-		RMM_TRY( RMM_FREE( h_str_cols [stringColCount], 0 ) );
+			NVStrings* const stringCol = NVStrings::create_from_index(h_str_cols[stringColCount],size_t(raw_csv->num_records));
+			if ((raw_csv->quotechar != '\0') && (raw_csv->doublequote==true)) {
+				// In PANDAS, default of enabling doublequote for two consecutive
+				// quotechar in quote fields results in reduction to single
+				std::string quotechar = std::string(&raw_csv->quotechar);
```

Comment:
Perhaps I'm missing something, but I believe it would be more efficient to just construct `quotechar` directly instead of creating a temporary, copying the temporary to `quotechar`, and then having to cleanup the temporary.
```suggestion
				std::string quotechar{&raw_csv->quotechar};
```

Reply:
Hm, that is surprising. I believe in this case the compiler is being smart and optimizing out the temporary.

In general, it's always better to prefer direct initialization vs. assignment because you're not guaranteed to get this optimization.

`std::string a{"asdf"} // initialization`
`std::string a = std::string{"asdf"} // assignment`